jujugui/static/gui/src/app/models/handlers.js

```diff
@@ -179,6 +246,23 @@ YUI.add('juju-delta-handlers', function(Y) {
         // attribute could be undefined.
         subordinate: change.Subordinate
       };
+      // Juju 2.0 changes the delta structure by removing Status, StatusInfo,
+      // and StatusData in favour of JujuStatus.Message and JujuStatus.Data.
+      // If change.JujuStatus is not defined then we will use the old delta
+      // structure.
+      var jujuStatus = change.JujuStatus;
+      var workloadStatus = change.WorkloadStatus;
+      if (jujuStatus) {
+        unitData.agent_state = utils.translateToLegacyAgentState(
```

Comment:
This code does not follow the logic described in the spec document.
```
If WorkloadStatus.Current == StatusError (“error”):
	Status = WorkloadStatus.Current
	StatusInfo = WorkloadStatus.Message
	StatusData = WorkloadStatus.Data
Else:
	Status = translate status with translation function at [1]
	StatusInfo = JujuStatus.Message
	StatusData = JujuStatus.Data
```
So this should be something like
```javascript
var jujuStatus = change.JujuStatus;
if (jujuStatus) {
  var workloadStatus = change.WorkloadStatus;
  if (workloadStatus.Current === statusError) {
    unitData.agent_state = workloadStatus.Current;
    unitData.agent_state_info = workloadStatus.Message;
    unitData.agent_state_data = workloadStatus.Data;
  } else {
    unitData.agent_state = utils.translateToLegacyAgentState(
      jujuStatus.Current, workloadStatus.Current, workloadStatus.Message);
    unitData.agent_state_info = jujuStatus.Message;
    unitData.agent_state_data = jujuStatus.Data;
  }
}
```